nuitka/tools/specialize/templates_python/StrNodes.py.j2

```diff
@@ -0,0 +1,137 @@
+class ExpressionStrOperation{{ operation_name }}(
+    {% if inheritance_base %}
+    ExpressionStrOperation{{ base_name }}Base
+    {% else%}
+    Expression{{ shape }}ShapeExactMixin,
+    ExpressionChildrenHavingBase
+    {% endif %}
+):
+    {% if not is_base%}
+    """This operation represents s.{{ operation_name.lower() }}({{arg_name}})."""
+
+    kind = "EXPRESSION_STR_OPERATION_{{ operation_name.upper() }}"
+    {% endif %}
+    {% if not inheritance_base %}
+    named_children = ("str_arg", "{{ arg_name.lower() }}")
+    {% endif %}
+    {% if simulator %}
+    {% if inheritance_base %}
+    @staticmethod
+    {% else%}
+    @abstractmethod
+    {% endif%}
+    def getSimulator({% if not inheritance_base %}self{% endif%}):
+        """Compile time simulation"""
+        {% if simulator_return_value != None %}
+        return {{ simulator_return_value }}
+        {% endif%}
+    {% endif %}
+    {% if not inheritance_base%}
+    def __init__(self, str_arg, {{ arg_name }}, source_ref):
+        assert str_arg is not None
+        assert {{ arg_name }} is not None
+
+        ExpressionChildrenHavingBase.__init__(
+            self,
+            values={"str_arg": str_arg, "{{ arg_name }}": {{ arg_name }}},
+            source_ref=source_ref,
+        )
+
+    def computeExpression(self, trace_collection):
+        str_arg = self.subnode_str_arg
+        {{ arg_name }} = self.subnode_{{ arg_name }}
+        if str_arg.isCompileTimeConstant() and {{ arg_name }}.isCompileTimeConstant():
+        {% if operation_name == "Strip2Base" %}
+            simulator = self.getSimulator()
+
+            return trace_collection.getCompileTimeComputationResult(
+                node=self,
+                computation=lambda: simulator(
+                    str_arg.getCompileTimeConstant(),
+                    chars.getCompileTimeConstant(),
+                ),
+                description="Built-in 'str.strip' with constant values.",
+                user_provided=str_arg.user_provided,
+            )
+
+        # TODO: Only if the sep is not a string
+        trace_collection.onExceptionRaiseExit(BaseException)
+
+        return self, None, None
+
+    def mayRaiseException(self, exception_type):
+        # TODO: Only if chars is not correct shape
+        return True
+
+        {% else %}
+            return trace_collection.getCompileTimeComputationResult(
+                node=self,
+                computation=lambda: str_arg.getCompileTimeConstant().{{ operation_name.lower() }}(
+                    {{ arg_name }}.getCompileTimeConstant()
+                ),
+                description="Built-in 'str.{{ operation_name.lower() }}' with constant values.",
+                user_provided=str_arg.user_provided,
+            )
+
+        trace_collection.onExceptionRaiseExit(BaseException)
+
+        return self, None, None
+        {% endif %}
+    {% endif %}
+    {% if operation_name in ["Partition", "Rpartition"]%}
+    @staticmethod
```

Comment:
Some sort of description of the operation should exist somewhere and be available in the template and say e.g. iterable result or not, and if static length, what it is, and maybe even the type shape. I saw the `shape` template arg, that's fine, but this can get quickly too many things, puting it is a dict by the operation name makes it easy for dumping related information.